Update Schemas documentation (#258)
CloudCannon/platform-documentation: articles/create-a-schema.mdx

  - defining-your-collections
related_links:
---
## What is a schema?

A schema is a predefined template that determines what should populate a [collection](/documentation/articles/define-your-collections/) file. You can configure several schemas for each collection, creating a list of file templates for you and your team.

> Schemas are excellent for creating files with consistent content and populating the initial contents of a new file. Schemas can contain front matter and Markdown content. When you use a schema to populate a new file, you can be sure that the inputs you need are available in the sidebar of the [Visual](/documentation/articles/introducing-the-visual-editor/) or [Content Editor](/documentation/articles/introducing-the-content-editor/).

Reviewer: ```suggestion
Schemas are excellent for creating files with consistent content and populating the initial contents of a new file. Schemas can apply to structured, markup, and combination files by containing both front matter and markup content or only data or markup content. When you use a schema to populate a new file, you can be sure that the inputs you need are available in the sidebar of the [Visual](/documentation/articles/introducing-the-visual-editor/) or [Content Editor](/documentation/articles/introducing-the-content-editor/).
```

We have also used the `name` key to modify the name of the schema in the *\+ Add* button dropdown menu. You can use the `name` key to provide context if you use the same schema in different collections.

<comp.DocsImage path="https://cc-dam.imgix.net/documentation/images/editing/schemas/2024-03/CloudCannon-Documentation-Add-Button-Multiple-Schemas.png" alt="A closeup of the Collection Browser shows multiple options in the Add dropdown." title="Multiple schemas in the collection Add menu" type="ui-snippet"/>

<comp.Notice info_type="info">
  Schemas create new files with the same file extension as the schema file. If you need to, you can reference schemas with different file extensions from the same collection to create different file types.
</comp.Notice>

## Update an existing schema

Schemas help you keep your collection files consistent. When you edit a file in a collection, CloudCannon checks that the inputs in the front matter match its defined schema. Updating a schema file by adding, removing, or reordering inputs will also update your collection files the next time you open them in an editing interface.

> <comp.Notice info_type="info">
>   CloudCannon does not update a collection file to match your schema until you open it in an editing interface.
> </comp.Notice>

Reviewer 2: ```suggestion
<comp.Notice info_type="info">
  CloudCannon does not update all existing collection files automatically when a schema changes. Each collection file will be updated when it is next opened in a non-source editing interface.
</comp.Notice>
```

Mainly trying to reword away from the ambiguous `it` (could read the original as `open it` referring to opening the schema in an editing interface, perhaps)

Let’s go over an example.

> We run a local news website. Our news collection has a default schema with inputs for each post's category, author, and title. Here is our default schema and a news post:

Reviewer 2: ```suggestion
We run a local news website. Our news collection has a default schema with inputs for each post's category, author, and title. Here is our default schema and an existing news post:
```

In this example we want to add the input for `image` to our default schema, reorder our inputs so that `title` appears before `author`, and remove the input for `category`.

<comp.CodeBlock language="Markdown" source="post.md">
```
---
title: My updated post
author:
image: /images/posts/header.png
---
Add news story here...
```
</comp.CodeBlock>

<comp.CodeBlock language="Markdown" source="holding_breath_record.md">
```
---
_schema: default
title: Local woman breaks the world record
author: John Simms
image: /images/posts/header.png
---
Today, Irma Fisher broke the world record for holding her breath.

Ms Fisher, who has lived in the local area since birth, credits her achievement to swimming practice in the nearby lake.
```
</comp.CodeBlock>

When we save our updates to the default schema and open our news story in an editing interface, we see that CloudCannon has updated the front matter to match our schema.

The inputs for `title`, `author`, and `image` now appear in the updated order, and `category` has been removed. Reordering the `author` key has not affected its value in the collection file.

<comp.Notice info_type="info">

>   You can customize this behavior using the `remove_extra_inputs`, `hide_extra_inputs`, `reorder_inputs` keys. For more information, please read our reference documentation for schemas.

Reviewer 2: Link?

>   You can customize this behavior using the `remove_extra_inputs`, `hide_extra_inputs`, `reorder_inputs` keys. For more information, please read our reference documentation for schemas.

Reviewer: ```suggestion
  You can customize this behavior using the `remove_extra_inputs`, `hide_extra_inputs`, `reorder_inputs` keys. For more information, please read our reference documentation for [schemas](/documentation/articles/schemas-reference/).
```

</comp.Notice>

> <comp.Notice info_type="important">
>   If you change the type of data stored in an input (e.g., switch from an object to a string or a string to a number), placeholder values in your schema may overwrite the values in your collection file.
> </comp.Notice>

Reviewer 2: ```suggestion
<comp.Notice info_type="important">
  If you change the type of data stored in an input (e.g., switch from an object to a string or a string to a number), placeholder values in your schema will overwrite the values in your collection file.
</comp.Notice>
```
